app/components/article-content.js

```diff
@@ -607,67 +608,76 @@ export default Component.extend(
 
     /**
      * Injects an affiliate unit into the article content
-     * @param {string} title
      */
-    handleBigAffiliateUnit(title) {
-      const unit = this.affiliateSlots.getBigUnitOnPage(title);
-
-      if (typeof unit === 'undefined') {
-        // There's no unit to display (not an error)
-        return;
-      }
+    handleBigAffiliateUnit() {
+      const indexForUnit = 1; // 2nd section
 
       // search for second section
       const h2Elements = this.element.querySelectorAll('h2[section]');
 
-      if (h2Elements[1]) {
-        const unitPlaceholder = document.createElement('div');
-        const unitWrapper = document.createElement('div');
+      // if there's no section that we want, just exit
+      if (!h2Elements[indexForUnit]) {
+        return;
+      }
+
+      this.affiliateSlots
+        .fetchUnitForPage(this.id, true)
+        .then(unit => {
+          if (typeof unit === 'undefined') {
+            // There's no unit to display (not an error)
+            return;
+          }
 
-        unitWrapper.appendChild(unitPlaceholder);
-        h2Elements[1].insertAdjacentElement('beforebegin', unitWrapper);
+          const unitPlaceholder = document.createElement('div');
+          const unitWrapper = document.createElement('div');
+          unitWrapper.appendChild(unitPlaceholder);
+          h2Elements[indexForUnit].insertAdjacentElement('beforebegin', unitWrapper);
 
-        this.renderedComponents.push(this.renderComponent({
-          name: 'affiliate-unit',
-          attrs: unit,
-          element: unitPlaceholder,
-        }));
-      }
+          this.renderedComponents.push(this.renderComponent({
```

Comment:
Is this spacing correct? Looks weird in the PR

Reply:
It is correct in the source file as - at least in ide